subiquity/models/subiquity.py

```diff
@@ -263,6 +275,46 @@ def is_postinstall_only(self, model_name):
     def confirm(self):
         self._confirmation.set()
 
+    def validate_cloudconfig_schema(self, data: dict, data_source: str):
+        """Validate data config adheres to strict cloud-config schema
+
+        Log warnings on any deprecated cloud-config keys used.
+
+        :param data: dict of valid cloud-config
+        :param data_source: str to present in logs/errors describing
+            where this config came from: autoinstall.user-data or system info
+
+        :raise SchemaValidationError: on invalid cloud-config schema
+        """
+        # cloud-init v. 22.3 will allow for log_deprecations=True to avoid
+        # raising errors on deprecated keys.
+        # In the meantime, iterate over schema_deprecations to log warnings.
+        try:
+            validate_cloudconfig_schema(data, schema=get_schema(), strict=True)
+        except SchemaValidationError as e:
+            if hasattr(e, "schema_deprecations"):
```

Comment:
Original release of bionic/focal or Jammy will not have schema_deprecations attr on SchemaValidationError. so be defensive about presence of that attribute in the even that subquity is trying to install from orig release images.